walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -1,23 +1,72 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public final class FindMeetingQuery {
+
+  private static final int MINUTES_IN_DAY = 60 * 24;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Generate a list of meeting actions (start or end) of the required attendees sorted by time.
+    // The order of 'start' and 'end' events with equal time permits 0 minute long meetings.
+    List<MeetingAction> sortedAttendeeEvents = events.stream()
+            .filter(event -> event.getAttendees().stream().anyMatch(request.getAttendees()::contains))
```

Comment:
You could probably use [Collections.disjoint](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collections.html#disjoint(java.util.Collection,java.util.Collection)) here:

```java
filter(event -> !Collections.disjoint(event.getAttendees(), request.getAttendees())) // Checks if two collections intersect.
```

Reply:
I updated the filtering predicate to make use of Collections.disjoint(c1, c2). 
In terms of the date/time handling, based on the classes we were provided, I believe we are expected to just use minutes as integers, but I'll take your note into account, if I work on a real project which will require time management.